ut/alarm_tables_testing.cpp

```diff
@@ -0,0 +1,321 @@
+
+  // Calculates which column should contain the alarm description based off the
+  // column number definitions within alarm_active_table.hpp  
+  int colnum_description = COLUMN_ALARMACTIVEDESCRIPTION - alarmActiveTable_COL_MIN;
+  // Checks that the alarm description column contains latest severity data.
+  std::size_t found = entries[colnum_description].find("Test alarm major raised description");
+  ASSERT_NE(std::string::npos, found);
+
+  // Clears the alarm and checks there are now no entries in the table.
+  alarmActiveTable_trap_handler(def_cleared);
+  std::vector<std::string> entries_update = snmp_walk(alarmActiveTable_table);
+  ASSERT_EQ(0, entries_update.size());
+}
+
+TEST_F(SNMPTest, ActiveTableIndex)
+{
+  // Creates three custom alarm definitions and raises the alarms.
+  AlarmDef::SeverityDetails cleared1(AlarmDef::CLEARED, "First alarm cleared description", "First alarm cleared details");
+  AlarmDef::SeverityDetails raised1(AlarmDef::CRITICAL, "First alarm raised description", "First alarm raised details");
+  AlarmDef::AlarmDefinition example1(6666, AlarmDef::SOFTWARE_ERROR, {cleared1, raised1});
+  AlarmTableDef def1_cleared(example1, cleared1);
+  AlarmTableDef def1_raised(example1, raised1);
+  alarmActiveTable_trap_handler(def1_raised);
+
+  AlarmDef::SeverityDetails cleared2(AlarmDef::CLEARED, "Second alarm cleared description", "Second alarm cleared details");
+  AlarmDef::SeverityDetails raised2(AlarmDef::CRITICAL, "Second alarm raised description", "Second alarm raised details");
+  AlarmDef::AlarmDefinition example2(6667, AlarmDef::SOFTWARE_ERROR, {cleared2, raised2});
+  AlarmTableDef def2_cleared(example2, cleared2);
+  AlarmTableDef def2_raised(example2, raised2);
+  alarmActiveTable_trap_handler(def2_raised);
+
+  AlarmDef::SeverityDetails cleared3(AlarmDef::CLEARED, "Third alarm cleared description", "Third alarm cleared details");
+  AlarmDef::SeverityDetails raised3(AlarmDef::CRITICAL, "Third alarm raised description", "Third alarm raised details");
+  AlarmDef::AlarmDefinition example3(6668, AlarmDef::SOFTWARE_ERROR, {cleared3, raised3});
+  AlarmTableDef def3_cleared(example3, cleared3);
+  AlarmTableDef def3_raised(example3, raised3);
+  alarmActiveTable_trap_handler(def3_raised);
+
+  // Shells out to snmpwalk to find all entries in Alarm Active Table.
+  std::vector<std::string> entries = snmp_walk(alarmActiveTable_table);
+
+  
+  // Retrieves the current date and time.
+  struct timespec ts;
+  struct tm timing;
+  time_t rawtime;
+  clock_gettime(CLOCK_REALTIME, &ts);
+  rawtime = ts.tv_sec;
+  timing = *localtime(&rawtime);
+  int day = timing.tm_mday;
+  int month = 1 + timing.tm_mon;
+
+  // Splits each part of the index field (separated by ".") for the first alarm into a vector.
+  std::vector<std::string> index_fields1;
+  Utils::split_string(entries[0], '.', index_fields1);
+
+  // As defined in RFC3877 the 17th and 16th elements of the index field are the day and
```

Comment:
Is it clearer to split this into two tests - one checking the timestamp, the other checking that the index is increasing?